### PR TITLE
Expose additional WinRM options for transport, basic auth, and SSPI 

### LINKED ISSUE #391: Expose additional winrm options
### Description

On FIPS enabled clients the WinRM transport can only use `:ssl` **transport** and `basic_auth`. NTLM auth uses `MD4` is is not at all feasible.

 ### Train and Platform Version

All verisons

### Replication Case

Ruby just barks when using `MD5` in FIPS mode. `MD4` is not at all available. 

The only working config on FIPS machine:

```
require 'winrm'

opts =
    {
    transport: :ssl,
    endpoint: 'https://<some server>:5986/wsman',
    user: 'Administrator',
    password: '<redacted>,
    no_ssl_peer_verification: true,
    basic_auth_only: true
    }

conn = WinRM::Connection.new(opts)

shell = conn.shell(:powershell)
puts shell.run('ls').stdout
shell.close
```

### PR DESCRIPTION
**Pull request by frezbo**
Signed-off-by: Noel Georgi <git@frezbo.com>

Fixes: https://github.com/inspec/train/issues/391

### CODE REVIEW

**Comment by clintoncwolfe** [comment]
I'm all for making it easier to use transports in the real world, with the settings needed to do so.  However, I think we need to think carefully about how we go about this: we already have a proliferation of options.

#3661 proposes using the configuration file to store credential sets - allowing transports to have extensive options and arguments.  With #3661, you could have multiple credentials for different winrm targets, with each configured as it needs.  Adding new options to that would be a non-event; InSpec would simply pass all information along to Train.

However, that's not convenient for more casual use.  Command line args are really what's wanted.  We face a few challenges:

1. Organizing CLI opts effectively
1. Allowing Train plugins to dynamically configure InSpec's CLI option parser (Thor) 

The second point is a long-term problem - winrm is part of core Train for today, and we can bear some overtight coupling on that.

For the first issue, I propose that we prefix all winrm options with `--winrm-` - so this PR would add `--winrm-transport`, `--winrm-disable-sspi`, and `--winrm-basic-auth`.  We'd then also alias existing WinRM options to include that prefix, and at some (major version) point deprecate the "prefix-less" versions of the options.  We could also use the Thor `group` facility to improve help output. 

Obviously, a similar approach could be taken for other transports that expose options (most egregiously SSH).

**Comment by frezbo** [comment]
@clintoncwolfe Would just prefixing the three options I added with `winrm` would be enough to get this going forward? Do you want me to update the **train** PR too, https://github.com/inspec/train/pull/392?

**Comment by clintoncwolfe** [comment]
Yep, that'll do for this PR.  I'll put in a review.

File: lib/inspec/base_cli.rb [hunk 1]
```diff
@@ -69,6 +69,12 @@ def self.target_options # rubocop:disable MethodLength
         desc: 'Use SSL for transport layer encryption (WinRM).'
       option :self_signed, type: :boolean,
         desc: 'Allow remote scans with self-signed certificates (WinRM).'
+      option :transport, type: :string,
```
Comment (clintoncwolfe) [inline lib/inspec/base_cli.rb]: As discussed, please prefix these with `--winrm-`

**Review by clintoncwolfe — CHANGES_REQUESTED** [review]
As discussed, just looking for the prefix change; we'll also need to change the version pin on Train in the gemspec.

File: lib/inspec/base_cli.rb [hunk 2]
```diff
@@ -69,6 +69,12 @@ def self.target_options # rubocop:disable MethodLength
...
+        desc: 'Specify which transport to use, defaults to negotiate (WinRM).'
```
Comment (clintoncwolfe) [inline lib/inspec/base_cli.rb:73]: I suggest we add a list of valid transport names.  Also, the help text says it defaults to `:negotiate`, but the Thor option doesn't have a default (perhaps omission causes the WinRM library to negotiate - but if so, a comment would be good)

**Review by clintoncwolfe — CHANGES_REQUESTED** [review]

**Comment by frezbo** [comment]
@clintoncwolfe PR updated, I will update the **gemspec** once a new version of train is released.

File: lib/inspec/base_cli.rb (hunk 2 above)
Comment (frezbo) [reply lib/inspec/base_cli.rb:73]: resolved.

File: lib/inspec/base_cli.rb (hunk 1 above)
Comment (frezbo) [reply lib/inspec/base_cli.rb]: resolved.

**Comment by clintoncwolfe** [comment]
Train 1.6.3 has been released, with the new options included.

**Comment by frezbo** [comment]
@clintoncwolfe The tests fail due to the newer ruby versions packaging `bundler` along with it. I have also seen this locally. The `functional` tests pass locally.

**Review by clintoncwolfe — APPROVED** [review]
Thanks so much, @frezbo !

**Review by jerryaldrichiii — APPROVED** [review]
Great work @frezbo!

**Comment by frezbo** [comment]
👍

**Pull request merged**